packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/useVirtualizerScrollInstance.tsx

```diff
@@ -0,0 +1,27 @@
+import {createContext, useContext} from 'react'
+
+/**
+ * @internal
+ */
+export interface VirtualizerScrollInstance {
+  scrollElement: HTMLElement | null
+}
+
+/**
+ * This is used to store the reference to the scroll element for virtualizer
+ * @internal
+ */
+export const VirtualizerScrollInstanceContext = createContext<VirtualizerScrollInstance | null>(
+  null
+)
+
+/**
+ * @internal
+ */
+export function useVirtualizerScrollInstance(): VirtualizerScrollInstance {
+  const ref = useContext(VirtualizerScrollInstanceContext)
+  if (!ref) {
+    throw new Error('VirtualizerScrollInstance: missing context value')
+  }
```

Comment:
+1

Reply:
For arrays the scroll needs to be specific to the parent it is enclosed inside. If we have it as `document.body` then we would still see issues where the array is shown partial. I personally believe it's better to throw an error which is easier to see when debugging vs it selecting some random element. 

Also since the document panel is wrapped in the context that is the top most parent, if no child dialogs or popover wrap the context the topmost panel would be selected. Let me know if that makes sense or y'all think otherwise

Reply:
hmm it has been marked as internal usage so I am not sure if it is a breaking change but I would concur to @bjoerge @skogsmaskin

Reply:
Good point about it being `@internal`. Let's go ahead and merge then.

Reply:
🚢